cbtemulator: init at 1.22.0

flokli opened the pull request:
This packages cbtemulator, an in-memory emulator for Google Bigtable.

I added a small smoketest, setting up tables, inserting a value, and asserting we get it back.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

flokli committed:
cbtemulator: init at 1.22.0

This packages cbtemulator, an in-memory emulator for Google Bigtable.

I added a small smoketest, setting up tables, inserting a value, and
asserting we get it back.

ryantm commented:
Tried to build this and got this problem:
```
$ nix run github:nixos/nixpkgs?ref=pull/298044/head#cbtemulator -L
cbtemulator> Running phase: unpackPhase
cbtemulator> unpacking source archive /nix/store/fr9lkc63nviqcwy6s83i1pg53fqffh3h-source
cbtemulator> source root is source
cbtemulator> Running phase: patchPhase
cbtemulator> Running phase: updateAutotoolsGnuConfigScriptsPhase
cbtemulator> Running phase: configurePhase
cbtemulator> Running phase: buildPhase
cbtemulator> go: 'go mod vendor' cannot be run in workspace mode. Run 'go work vendor' to vendor the workspace or set 'GOWORK=off' to exit workspace mode.
error: builder for '/nix/store/l2kx177461hmki26ir5qz327ys71hxql-cbtemulator-1.22.0-go-modules.drv' failed with exit code 1;
```

flokli commented:
@ryantm I don't quite get it. I first assumed some cached FOD output on my side, removed `modRoot` and `goWork`, and added  `/bigtable` , but it came up with the same `vendorHash`, and builds on my machne (x86_64-linux). Does it still fail for you now?

sikmir commented on pkgs/by-name/cb/cbtemulator/package.nix:
```diff
@@ -0,0 +1,65 @@
+    repo = "google-cloud-go";
+    rev = "bigtable/v${version}";
+    hash = "sha256-eOi4QFthnmZb5ry/5L7wzr4Fy1pF/H07BzxOnXtmSu4=";
+  }) + "/bigtable";
```
What about  `sourceRoot`?

flokli replied:
```
building '/nix/store/j46p6z7pyvwxzwcc992w5kmsldrw2xmb-cbtemulator-1.22.0-go-modules.drv' on 'ssh://root@box.flokli.de'...
copying 0 paths...
Running phase: unpackPhase
unpacking source archive /nix/store/fr9lkc63nviqcwy6s83i1pg53fqffh3h-source
source root is bigtable
chmod: cannot access 'bigtable': No such file or directory
```

flokli replied:
I added a comment.

cbrewster commented:
It builds for me, I also tried recalculating the `vendorHash` and it matches what is shown in this PR. :thinking:

flokli commented on pkgs/by-name/cb/cbtemulator/package.nix:
```diff
@@ -0,0 +1,65 @@
+      ${lib.getExe pkgs.cbtemulator} &
+
+      cleanup() {
+        pkill cbtemulator
```
Note to self: this doesn't seem to bring pkill into $PATH, check Darwin manually

flokli replied:
`procps` doesn't have `pkill` on MacOS, but I rewrote this to just use the builtin `kill` bash builtin.

ryantm commented:
> Does it still fail for you now?

Strange. It builds for me fine now!

flokli commented:
Fixed the smoketest on aarch64-darwin.

sikmir commented:
Result of `nixpkgs-review pr 298044` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>cbtemulator</li>
  </ul>
</details>

sikmir commented on pkgs/by-name/cb/cbtemulator/package.nix:
```diff
@@ -0,0 +1,67 @@
+        nativeBuildInputs = [ google-cloud-bigtable-tool ];
+      } ''
+      # Start the emulator
+      ${lib.getExe pkgs.cbtemulator} &
```
```suggestion
      ${lib.getExe cbtemulator} &
```

flokli replied:
Done

sikmir commented on pkgs/by-name/cb/cbtemulator/package.nix:
```diff
@@ -0,0 +1,67 @@
+{ pkgs
```
```suggestion
{ cbtemulator
```

flokli replied:
Done
